test/end2end_test.go

```diff
@@ -2281,6 +2289,56 @@ func testClientRequestBodyErrorCancelStreamingInput(t *testing.T, e env) {
 	})
 }
 
+func TestFailFastRPCRetryOnCredentialTimeout(t *testing.T) {
+	const minConnectTimeout = time.Millisecond
+	defer internal.TestingSetMinConnectTimeout(minConnectTimeout)()
+
+	before := grpc.DefaultBackoffConfig.BaseDelay
+	grpc.DefaultBackoffConfig.BaseDelay = 100 * time.Millisecond
+	defer func() {
+		grpc.DefaultBackoffConfig.BaseDelay = before
+	}()
+
+	te := newTest(t, env{name: "slow-tls", network: "tcp", security: "tls"})
```

Comment:
If that's really a concern, minConnectTimeout should move to a field on the options struct.